src/simpleclient.ts

```diff
@@ -0,0 +1,561 @@
+        );
+        const uiDepositDisplay = ceilToDecimal(uiDeposit, decimals);
+        const uiBorrow = marginAccount.getUiBorrow(this.mangoGroup, tokenIndex);
+        const uiBorrowDisplay = ceilToDecimal(uiBorrow, decimals);
+
+        return new MarginAccountBalance(
+          tokenSymbol,
+          uiDepositDisplay,
+          uiBorrowDisplay,
+        );
+      });
+
+      const marketBalances = this.mangoGroup.spotMarkets.map((marketPk) => {
+        const market = this.markets.find((market) =>
+          market.publicKey.equals(marketPk),
+        );
+        if (market === undefined) {
+          throw new Error(`market for ${marketPk.toBase58()} not found`);
+        }
+
+        const token = Object.entries(this.mangoGroupTokenMappings).find(
+          (entry) => {
+            return entry[1].equals(market.baseMintAddress);
+          },
+        );
+
+        const tokenIndex = this.mangoGroup.getTokenIndex(
+          market.baseMintAddress,
+        );
+        const openOrders: OpenOrders =
+          marginAccount.openOrdersAccounts[tokenIndex]!;
+        const nativeBaseFree = openOrders?.baseTokenFree || new BN(0);
+        const nativeBaseLocked = openOrders
+          ? openOrders.baseTokenTotal.sub(nativeBaseFree)
+          : new BN(0);
+        const nativeBaseUnsettled = openOrders?.baseTokenFree || new BN(0);
+        const orders = nativeToUi(
+          nativeBaseLocked.toNumber(),
+          this.mangoGroup.mintDecimals[tokenIndex],
+        );
+        const unsettled = nativeToUi(
+          nativeBaseUnsettled.toNumber(),
+          this.mangoGroup.mintDecimals[tokenIndex],
+        );
+
+        const quoteToken = Object.entries(this.mangoGroupTokenMappings).find(
+          (entry) => {
+            return entry[1].equals(market.quoteMintAddress);
+          },
+        );
+        const quoteCurrencyIndex = this.mangoGroup.getTokenIndex(
+          market.quoteMintAddress,
+        );
+        const nativeQuoteFree = openOrders?.quoteTokenFree || new BN(0);
+        const nativeQuoteLocked = openOrders
+          ? openOrders!.quoteTokenTotal.sub(nativeQuoteFree)
+          : new BN(0);
+        const nativeQuoteUnsettled = openOrders?.quoteTokenFree || new BN(0);
+        const ordersQuote = nativeToUi(
+          nativeQuoteLocked.toNumber(),
+          this.mangoGroup.mintDecimals[quoteCurrencyIndex],
+        );
+        const unsettledQuote = nativeToUi(
+          nativeQuoteUnsettled.toNumber(),
+          this.mangoGroup.mintDecimals[quoteCurrencyIndex],
+        );
+
+        return new MarketBalance(
+          token![0],
+          orders,
+          unsettled,
+          quoteToken![0],
+          ordersQuote,
+          unsettledQuote,
+        );
+      });
+
+      return new Balance(
+        marginAccount.publicKey,
+        marginAccountBalances,
+        marketBalances,
+      );
+    });
+  }
+
+  async getPnl() {
+    // todo
+    throw new Error('not implemented');
```

Comment:
We use another service, that is not open source yet, but provides you with a trade history for a specific open orders account. You can see how to use it here: https://github.com/blockworks-foundation/mango-ui/blob/5715e09fee5f5f606a8c86f72f1a622f6307858d/src/utils/useTradeHistory.tsx#L36

And here is code for the PnL (unrealized) calculation https://github.com/blockworks-foundation/mango-ui-v2/blob/5f61a86a615e4c92f53d565f81de1622c615f365/components/MarginInfo.tsx#L11

Reply:
done, also added tests